aarch64-darwin fail with only in github-actions:
```
error: builder for '/nix/store/szm60mz1vm2x0z000jhf8wd2inyq5ifl-python3.12-unicorn-2.1.1.drv' failed with exit code 132;
       last 25 log lines:
       > checking for references to /private/tmp/nix-build-python3.12-unicorn-2.1.1.drv-0/ in /nix/store/ihhiw8i92vdvd84v2f9nbzic2cm66xkp-python3.12-unicorn-2.1.1...
       > patching script interpreter paths in /nix/store/ihhiw8i92vdvd84v2f9nbzic2cm66xkp-python3.12-unicorn-2.1.1
       > stripping (with command strip and flags -S) in  /nix/store/ihhiw8i92vdvd84v2f9nbzic2cm66xkp-python3.12-unicorn-2.1.1/lib
       > checking for references to /private/tmp/nix-build-python3.12-unicorn-2.1.1.drv-0/ in /nix/store/h87ycmybj67shq9hc82psqxnfpyghbdf-python3.12-unicorn-2.1.1-dist...
       > patching script interpreter paths in /nix/store/h87ycmybj67shq9hc82psqxnfpyghbdf-python3.12-unicorn-2.1.1-dist
       > Executing pythonRemoveTestsDir
       > Finished executing pythonRemoveTestsDir
       > Running phase: installCheckPhase
       > patching script interpreter paths in sample_arm.py sample_arm64.py sample_arm64eb.py sample_armeb.py sample_ctl.py sample_m68k.py sample_mips.py sample_network_auditing.py sample_ppc.py sample_riscv.py sample_s390x.py sample_sparc.py sample_tricore.py sample_x86.py shellcode.py
       > sample_arm.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_arm64.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_arm64eb.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_armeb.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_ctl.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_m68k.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_mips.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_network_auditing.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_ppc.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_riscv.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_sparc.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_tricore.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > sample_x86.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > shellcode.py: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/bin/python"
       > Emulate ARM code
       > sample_all.sh: line 3: 11445 Illegal instruction: 4  python3 ./sample_arm.py
       For full logs, run 'nix log /nix/store/szm60mz1vm2x0z000jhf8wd2inyq5ifl-python3.12-unicorn-2.1.1.drv'.
 ```
 
 
 psutil build broke for darwin:
 ```
 error: builder for '/nix/store/6l5i401bfjr6q36i8irnfp4a61z7xpl7-python3.12-psutil-5.9.8.drv' failed with exit code 1;
       last 25 log lines:
       >           u_char  sdl_alen;       /* link level address length */
       >           ^
       >   /nix/store/hzxiynjmmj35fpy3jla7vcqwmzj9i449-Libsystem-1238.60.2/include/net/if_dl.h:96:2: error: unknown type name 'u_char'; did you mean 'char'?
       >           u_char  sdl_slen;       /* link layer selector length */
       >           ^
       >   psutil/arch/osx/net.c:68:21: warning: array subscript is of type 'char' [-Wchar-subscripts]
       >               ifc_name[sdl->sdl_nlen] = 0;
       >                       ^~~~~~~~~~~~~~
       >   1 warning and 7 errors generated.
       >   XCode (https://developer.apple.com/xcode/) is not installed
       >   error: command '/nix/store/kk0zkxvj9k09ra65gg9pzd9mlzykbxps-clang-wrapper-16.0.6/bin/clang' failed with exit code 1
       >   error: subprocess-exited-with-error
       >
       >   × Building wheel for psutil (pyproject.toml) did not run successfully.
       >   │ exit code: 1
       >   ╰─> See above for output.
       >
       >   note: This error originates from a subprocess, and is likely not a problem with pip.
       >   full command: /nix/store/ks8acr22s4iggnmvxydm5czl30racy32-python3-3.12.5/bin/python3.12 /nix/store/g24k5pcsml27wrdbg0cnv4z43sg1x2j0-python3.12-pip-24.0/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /private/tmp/nix-build-python3.12-psutil-5.9.8.drv-0/tmprpjqaw0o
       >   cwd: /private/tmp/nix-build-python3.12-psutil-5.9.8.drv-0/psutil-5.9.8
       >   Building wheel for psutil (pyproject.toml) ... error
       >   ERROR: Failed building wheel for psutil
       > Failed to build psutil
       > ERROR: Failed to build one or more wheels
       > 
       For full logs, run 'nix log /nix/store/6l5i401bfjr6q36i8irnfp4a61z7xpl7-python3.12-psutil-5.9.8.drv'.
```

--- a/.github/workflows/lock.yml
+++ b/.github/workflows/lock.yml
@@ -16,42 +16,87 @@ on:
       - '!*.md'
 
 jobs:
-  check_release_build-gdb-x86_64:
-    runs-on: ubuntu-latest
+  check_release_build-gdb:
+    strategy:
+      fail-fast: false
+      matrix:
+        # https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+        os: [
+          ubuntu-latest,  # x86_64-linux
+          # ubuntu-arm,  # aarch64-linux, not public yet
+        ]
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
       with:
         nix_path: nixpkgs=channel:nixos-unstable
 
     - name: build pwndbg
-      run: nix build '.#pwndbg' -o result1
+      run: nix build '.#pwndbg' --accept-flake-config -o result
 
     - name: simple run pwndbg
-      run: ./result1/bin/pwndbg <<< 'exit'
+      run: ./result/bin/pwndbg <<< 'exit'
 
-  check_release_build-lldb-x86_64:
-    runs-on: ubuntu-latest
+    - name: configure cache
+      if: github.ref == 'refs/heads/dev'
+      uses: cachix/cachix-action@ad2ddac53f961de1989924296a1f236fcfbaa4fc  # v15
+      with:
+        name: pwndbg
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+        skipPush: true
+
+    - name: upload cache
+      if: github.ref == 'refs/heads/dev'
+      run: cachix push -v pwndbg ./result
+
+  check_release_build-lldb:
+    strategy:
+      fail-fast: false
+      matrix:
+        # https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+        os: [
+          ubuntu-latest,  # x86_64-linux
+          # ubuntu-arm,  # aarch64-linux, not public yet
+          macos-13,  # x86_64-darwin
+          macos-15,  # aarch64-darwin
+        ]
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
       with:
         nix_path: nixpkgs=channel:nixos-unstable
 
     - name: build pwndbg
-      run: nix build '.#pwndbg-lldb' -o result2
+      run: nix build '.#pwndbg-lldb' --accept-flake-config -o result
 
     - name: simple run pwndbg
-      run: ./result2/bin/pwndbg-lldb <<< 'exit'
+      run: ./result/bin/pwndbg-lldb <<< 'exit'
+
+    - name: configure cache
+      if: github.ref == 'refs/heads/dev'
+      uses: cachix/cachix-action@ad2ddac53f961de1989924296a1f236fcfbaa4fc  # v15
+      with:
+        name: pwndbg
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+        skipPush: true
+
+    - name: upload cache
+      if: github.ref == 'refs/heads/dev'
+      run: cachix push -v pwndbg ./result
+
 
   lock_flake:
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
       with:
         nix_path: nixpkgs=channel:nixos-unstable
 
@@ -63,7 +108,7 @@ jobs:
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
       with:
         nix_path: nixpkgs=channel:nixos-unstable
 

--- a/.github/workflows/releases.yml
+++ b/.github/workflows/releases.yml
@@ -17,7 +17,7 @@ jobs:
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
       with:
         nix_path: nixpkgs=channel:nixos-unstable
 
@@ -55,7 +55,7 @@ jobs:
 #    timeout-minutes: 60
 #    steps:
 #    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9  # @v3
-#    - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac  # @v22
+#    - uses: cachix/install-nix-action@08dcb3a5e62fa31e2da3d490afc4176ef55ecd72  # @v30
 #      with:
 #        nix_path: nixpkgs=channel:nixos-unstable
 #

--- a/flake.nix
+++ b/flake.nix
@@ -1,6 +1,15 @@
 {
   description = "pwndbg";
 
+  nixConfig = {
+    extra-substituters = [
+      "https://pwndbg.cachix.org"
+    ];
+    extra-trusted-public-keys = [
+      "pwndbg.cachix.org-1:HhtIpP7j73SnuzLgobqqa8LVTng5Qi36sQtNt79cD3k="
+    ];
+  };
+
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   inputs.poetry2nix = {
     url = "github:nix-community/poetry2nix";

--- a/nix/pyenv.nix
+++ b/nix/pyenv.nix
@@ -8,14 +8,13 @@
   ...
 }:
 pkgs.poetry2nix.mkPoetryEnv {
-  groups = lib.optionals isDev [ "dev" ] ++ lib.optionals isLLDB [ "lldb" ];
+  groups = [ "main" ] ++ lib.optionals isDev [ "dev" ] ++ lib.optionals isLLDB [ "lldb" ];
   checkGroups = lib.optionals isDev [ "dev" ] ++ lib.optionals isLLDB [ "lldb" ];
   projectDir = inputs.pwndbg;
   python = python3;
   overrides = pkgs.poetry2nix.overrides.withDefaults (
     self: super: {
       pip = python3.pkgs.pip; # fix infinite loop in nix, look here: https://github.com/nix-community/poetry2nix/issues/1184#issuecomment-1644878841
-      unicorn = python3.pkgs.unicorn; # fix build for aarch64 (but it will use same version like in nixpkgs)
 
       # disable build from source, because rust's hash had to be repaired many times, see: PR https://github.com/pwndbg/pwndbg/pull/2024
       cryptography = super.cryptography.override { preferWheel = true; };
@@ -27,6 +26,31 @@ pkgs.poetry2nix.mkPoetryEnv {
       pt = super.pt.overridePythonAttrs (old: {
         buildInputs = (old.buildInputs or [ ]) ++ [ super.poetry-core ];
       });
+
+      # Patch psutil to work on macOS (Darwin)
+      # https://github.com/pwndbg/pwndbg/pull/2526#issuecomment-2476732310
+      psutil = (
+        super.psutil.overridePythonAttrs (
+          old:
+          pkgs.lib.optionalAttrs pkgs.stdenv.isDarwin {
+            stdenv = pkgs.overrideSDK pkgs.stdenv "11.0";
+            NIX_CFLAGS_COMPILE = "-DkIOMainPortDefault=0";
+            buildInputs =
+              old.buildInputs or [ ]
+              ++ pkgs.lib.optionals pkgs.stdenv.isx86_64 [ pkgs.darwin.apple_sdk.frameworks.CoreFoundation ]
+              ++ [ pkgs.darwin.apple_sdk.frameworks.IOKit ];
+          }
+        )
+      );
+
+      # Disable tests for unicorn on macOS in GitHub Actions (to avoid segmentation faults)
+      # https://github.com/pwndbg/pwndbg/pull/2526#issuecomment-2476732310
+      unicorn = python3.pkgs.unicorn.overridePythonAttrs (
+        old:
+        pkgs.lib.optionalAttrs pkgs.stdenv.isDarwin {
+          doCheck = false;
+        }
+      );
 
       capstone =
         # capstone=5.0.3 build is broken only in darwin :(, soo we use wheel

